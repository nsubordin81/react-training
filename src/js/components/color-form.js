import React from 'react';

export class ColorForm extends React.Component {

    static propTypes = {
        addColor: React.PropTypes.func.isRequired
    }

    constructor(props) {
        super(props);

        this.state = {
            newColor: ''
        };
    }

    onChange = (e) => {
        this.setState({
            [e.currentTarget.name]: e.currentTarget.value
        });
    };

    // addNewColor = () => {
    //     this.props.addColor(this.state.newColor);
    //     // this.setState({
    //     //     newColor: ''
    //     // });
    // };

    veryBadIdea = () => {
        console.log(this.input.value);
    };

    componentDidMount() {
        console.log(this.input);
        this.input.addEventListener('keyup', this.veryBadIdea);

        this.newColorInput.focus();
    }

    componentWillUnmount() {
        this.input.removeEventListener('keyup', this.veryBadIdea);
    }

    render() {

        const myDefaultValue = 'We do not want the Patriots to win the Super Bowl!';

        return <form>
            <div>
                <label htmlFor="new-color-input">New Color</label>
                <input type="text" id="new-color-input" name="newColor"
                    value={this.state.newColor} onChange={this.onChange}
                    ref={(input) => this.newColorInput = input} />
            </div>
            <div>
                <input type="text" defaultValue={myDefaultValue}
                    ref={(input) => { this.input = input; }} />
            </div>
            <button type="button" onClick={() => this.props.addColor(this.state.newColor)}>Add Color</button>
        </form>;
    }    

}
