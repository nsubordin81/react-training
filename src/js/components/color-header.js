import React from 'react';

export class ColorHeader extends React.Component {
    render() {

        const headerText = '<h1>Color Tool</h1>';

        return <div dangerouslySetInnerHTML={({ __html: headerText})}></div>;
    }
}