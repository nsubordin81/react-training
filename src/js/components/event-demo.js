import React from 'react';

export class EventDemo extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            counter: 1
        };
    }

    onClick = () => {
        this.setState({ counter: this.state.counter + 1 });
    }

    render() {
        return <div>
            <div>{this.state.counter}</div>
            <button onClick={this.onClick}>Increment</button>
        </div>;
    }

}