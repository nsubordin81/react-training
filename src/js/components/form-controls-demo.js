import React from 'react';
import ReactDOM from 'react-dom';

class DemoForm extends React.PureComponent {

    constructor(props) {
        super(props);

        this.state = {
            inputText: '',
            inputNumber: 0,
            inputDate: '2017-02-01',
            inputRange: 25,
            inputColor: '#f3a122',
            inputCheckbox: true,
            inputRadio: 'radio2',
            textarea: '',
            selectDropDown: '',
            selectListBox: '',
            selectMultiSelect: []
        };
    }

    onChange = e => {

        let controlValue;

        switch (e.currentTarget.type) {
            case 'number':
                controlValue = Number(e.currentTarget.value);
                break;
            case 'checkbox':
                controlValue = e.currentTarget.checked;
                break;
            default:

                if (e.currentTarget.multiple) {
                    controlValue = Array.from(e.currentTarget.options)
                        .filter(option => option.selected)
                        .map(option => option.value);

                } else {
                    controlValue = e.currentTarget.value;
               }

                break;
        }

        this.setState({
            [e.currentTarget.name]: controlValue
        });
    };

    render() {

        return <form>
            <div>
                <label>Input Text:</label>
                <input type="text" list="color-preferences" name="inputText" required value={this.state.inputText} onChange={this.onChange} />
                <br /><span>{this.state.inputText}, {typeof this.state.inputText}</span>
                <datalist id="color-preferences">
                    <option>Red</option>
                    <option>Blue</option>
                    <option>Brown</option>
                    <option>Black</option>
                </datalist>
            </div>
            <div>
                <label>Input Number:</label>
                <input type="number" name="inputNumber" value={this.state.inputNumber} onChange={this.onChange} />
                <br /><span>{this.state.inputNumber}, {typeof this.state.inputNumber}</span>
            </div>
            <div>
                <label>Input Date:</label>
                <input type="date" name="inputDate" value={this.state.inputDate} onChange={this.onChange} />
                <br /><span>{this.state.inputDate}, {typeof this.state.inputDate}</span>
            </div>
            <div>
                <label>Input Range:</label>
                <input type="range" name="inputRange" min="20" max="60" value={this.state.inputRange} onChange={this.onChange} />
                <br /><span>{this.state.inputRange}, {typeof this.state.inputRange}</span>
            </div>
            <div>
                <label>Input Color:</label>
                <input type="color" name="inputColor" value={this.state.inputColor} onChange={this.onChange} />
                <br /><span>{this.state.inputColor}, {typeof this.state.inputColor}</span>
            </div>
            <div>
                <label>Input Checkbox:</label>
                <input type="checkbox" name="inputCheckbox" checked={this.state.inputCheckbox} onChange={this.onChange} />
                <br /><span>{this.state.inputCheckbox ? 'true' : 'false'}, {typeof this.state.inputCheckbox}</span>
            </div>
            <fieldset>
                <legend>Keith is legend!</legend>
                <div>
                    <label>Input Radio 1:</label>
                    <input type="radio" name="inputRadio" checked={this.state.inputRadio === 'radio1'} value="radio1" onChange={this.onChange} />
                </div>                
                <div>
                    <label>Input Radio 2:</label>
                    <input type="radio" name="inputRadio" checked={this.state.inputRadio === 'radio2'} value="radio2" onChange={this.onChange} />
                </div>                
                <div>
                    <label>Input Radio 3:</label>
                    <input type="radio" name="inputRadio" checked={this.state.inputRadio === 'radio3'} value="radio3" onChange={this.onChange} />
                </div>                
                <span>{this.state.inputRadio}, {typeof this.state.inputRadio}</span>
            </fieldset>
            <div>
                <label>Textarea:</label>
                <textarea name="textarea" value={this.state.textarea} onChange={this.onChange} />
                <br /><span>{this.state.textarea}, {typeof this.state.textarea}</span>
            </div>            
            <div>
                <label>Drop Down:</label>
                <select name="selectDropDown" value={this.state.selectDropDown} onChange={this.onChange}>
                    <option value="">Select One...</option>
                    <option value="1">First One</option>
                    <option value="2">Second One</option>
                    <option value="3">Third One</option>
                </select>
                <br /><span>{this.state.selectDropDown}, {typeof this.state.selectDropDown}</span>
            </div>
            <div>
                <label>List Box:</label>
                <select name="selectListBox" size="5" value={this.state.selectListBox} onChange={this.onChange}>
                    <option value="1">First One</option>
                    <option value="2">Second One</option>
                    <option value="3">Third One</option>
                </select>
                <br /><span>{this.state.selectListBox}, {typeof this.state.selectListBox}</span>
            </div>            
            <div>
                <label>Multi-Select List Box:</label>
                <select name="selectMultiSelect" size="5" multiple value={this.state.selectMultiSelect} onChange={this.onChange}>
                    <option value="1">First One</option>
                    <option value="2">Second One</option>
                    <option value="3">Third One</option>
                </select>
                <br /><span>{JSON.stringify(this.state.selectMultiSelect)}, {typeof this.state.selectMultiSelect}</span>
            </div>            
        </form>;

    }


}


ReactDOM.render(<DemoForm />, document.querySelector('main'));
