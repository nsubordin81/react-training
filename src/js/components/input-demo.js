import React from 'react';

export class InputDemo extends React.Component {

    static propTypes = {
        message: React.PropTypes.string
    };

    constructor(props) {
        super(props);
        this.state = {
            message: props.message
        };
    }

    onChange = e => {
        this.setState({ [e.currentTarget.name]: e.currentTarget.value });
    }

    render() {
        return <form id="form">
            <label id="label">
                Message: <input name="message" value={this.state.message} onChange={this.onChange} />
            </label>
        </form>;
    }

}
