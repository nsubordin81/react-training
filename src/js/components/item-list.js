import React from 'react';

export class ViewItemRow extends React.Component {


    render() {
        return <li>
            {this.props.item}
            <button type="button" onClick={() => this.props.deleteRow(this.props.item)}>Delete</button>
            <button type="button" onClick={() => this.props.editRow(this.props.item)}>Edit</button>
        </li>;
    }

}

export class EditItemRow extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            item: props.item
        };
    }

    onChange = (e) => {
        this.setState({
            [e.currentTarget.name]: e.currentTarget.value
        });
    }

    render() {
        return <li>
            <input type="text" id="item-input" name="item" value={this.state.item} onChange={this.onChange} />
            <button type="button" onClick={() => this.props.saveRow(this.state.item)}>Save</button>
            <button type="button" onClick={() => this.props.cancelRow()}>Cancel</button>
        </li>;
    }

}

export class ItemList extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            editItem: ''
        };
    }

    static propTypes = {
        items: React.PropTypes.array
    };

    editRow = (item) => {
        this.setState({
            editItem: item
        });
    };

    deleteRow = (item) => {
        this.props.deleteRow(item);
    };

    saveRow = (item) => {
        this.props.saveRow(this.state.editItem, item);
        this.setState({
            editItem: ''
        });        
    };

    cancelRow = () => {
        this.setState({
            editItem: ''
        });
    };

    render() {
        return <ul>
            {this.props.items.map((item, index, list) =>
                this.state.editItem === item
                    ? <EditItemRow key={index} item={item} saveRow={this.saveRow} cancelRow={this.cancelRow} />
                    : <ViewItemRow key={index} item={item} editRow={this.editRow} deleteRow={this.deleteRow} />
            )}
        </ul>;
    }
}