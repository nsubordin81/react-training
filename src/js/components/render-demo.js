import React from 'react';

export class RenderDemo extends React.Component {
    
    render() {
        return <h1>Hello World!!</h1>;
    }
}