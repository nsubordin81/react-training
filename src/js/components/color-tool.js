import React from 'react';

import { ColorHeader as MyColorHeader } from './color-header'; 
import { ItemList } from './item-list';
import { ColorForm } from './color-form';

export class ColorTool extends React.Component {

    static propTypes = {
        myColors: React.PropTypes.array
    };    

    constructor(props) {
        super(props);

        console.log(props);

        this.state = {
            colors: this.props.myColors.concat()
        };
    }

    addColor = (newColor) => {
        this.setState({
            colors: this.state.colors.concat(newColor)
        });
    }

    deleteColor = (deleteColor) => {
        this.setState({
            colors: this.state.colors.filter(color => color !== deleteColor)
        });
    }

    saveColor = (oldColor, newColor) => {
        this.setState({
            colors: this.state.colors.filter(color => color !== oldColor).concat(newColor)
        });        
    }

    render() {

        return <div>
            <MyColorHeader />
            <ItemList items={this.state.colors} deleteRow={this.deleteColor} saveRow={this.saveColor} />
            <ColorForm addColor={this.addColor} />
        </div>;
    }
}