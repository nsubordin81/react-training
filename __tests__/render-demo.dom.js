import React from 'react';
import TestUtils from 'react-addons-test-utils';
import { RenderDemo } from '../src/js/components/render-demo';
import renderer from 'react-test-renderer';

jest.unmock('../src/js/components/render-demo');

describe('<RenderDemo /> Mock DOM', () => {

    let component;
    let componentDOMNode;

    beforeEach(() => {
        component = TestUtils.renderIntoDocument(<RenderDemo />);
        componentDOMNode = TestUtils.findRenderedDOMComponentWithTag(component, 'h1');
    });

    test('<RenderDemo /> renders', () => {
        expect(componentDOMNode.textContent).toBe('Hello World!!');
    });
	
});

describe('<RenderDemo /> Mock DOM', () => {

    test('<RenderDemo /> snapshot', () => {

        const tree = renderer.create(<RenderDemo />).toJSON();
        expect(tree).toMatchSnapshot();

    });


});