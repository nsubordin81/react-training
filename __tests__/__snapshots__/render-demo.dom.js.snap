exports[`<RenderDemo /> Mock DOM <RenderDemo /> snapshot 1`] = `
<h1>
  Hello World!!
</h1>
`;
